viamonvifdiscovery/service.go

```diff
@@ -74,10 +74,22 @@ func newDiscovery(_ context.Context, _ resource.Dependencies,
 }
 
 // DiscoverResources discovers different rtsp cameras that use onvif.
-func (dis *rtspDiscovery) DiscoverResources(ctx context.Context, _ map[string]any) ([]resource.Config, error) {
+func (dis *rtspDiscovery) DiscoverResources(ctx context.Context, extra map[string]any) ([]resource.Config, error) {
 	cams := []resource.Config{}
 
-	list, err := viamonvif.DiscoverCameras(ctx, dis.Credentials, nil, dis.logger)
+	discoverCreds := dis.Credentials
+	// check for a username from extras
+	extraUser, ok := extra["User"].(string)
+	if ok {
+		// not requiring a password to match config
+		extraPass, ok := extra["Pass"].(string)
+		if !ok {
+			extraPass = ""
+		}
+		addCred := device.Credentials{User: extraUser, Pass: extraPass}
+		discoverCreds = append(discoverCreds, addCred)
+	}
+	list, err := viamonvif.DiscoverCameras(ctx, discoverCreds, nil, dis.logger)
```

Comment:
If extra is nil I think the user check will panic. 
Let's:
1. Make this a helper function to return the creds to pass to viamonvif.DiscoverCamers.
2. Bail if extra is nil
3. Write a test to ensure you're getting what you want out of the reutns from the new function.

Reply:
nil didn't have an issue when testing in client code, are there other ways it could happen?

Reply:
also I can't do the tests I want until I can mock cameras, which won't be until after we move into the `viamonvif` folder

Reply:
Can add the helper + test it tho

Reply:
Add the tst - I wonder if we're passing through an an empty map[string]interface{} at this point to be defensive.
But no harm in being extra defensive